drivers/gpio/gpio_eos_s3.c

```diff
@@ -0,0 +1,311 @@
+		gpio_cfg.xPadConf->ucSmtTrg = PAD_SMT_TRIG_EN;
+	}
+
+	if ((flags & GPIO_OUTPUT) != 0) {
+		if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0) {
+			MISC_CTRL->IO_OUTPUT |= (BIT(gpio_num) & GPIOS_MASK);
+		} else if ((flags & GPIO_OUTPUT_INIT_LOW) != 0) {
+			MISC_CTRL->IO_OUTPUT &= ~(BIT(gpio_num) & GPIOS_MASK);
+		}
+		gpio_cfg.xPadConf->ucMode = PAD_MODE_OUTPUT_EN;
+	}
+
+	/* Initial PAD configuration */
+	HAL_PAD_Config(gpio_cfg.xPadConf);
+
+	/* Override direction setup to support bidirectional config */
+	if ((flags & GPIO_DIR_MASK) == (GPIO_INPUT | GPIO_OUTPUT)) {
+		io_mux += gpio_cfg.xPadConf->ucPin;
+		*io_mux &= ~PAD_OEN_DISABLE;
+		*io_mux |= PAD_REN_ENABLE;
+	}
+
+	return 0;
+}
+
+static int gpio_eos_s3_port_get_raw(const struct device *dev,
+				    uint32_t *value)
+{
+	ARG_UNUSED(dev);
+
+	*value = (MISC_CTRL->IO_INPUT & GPIOS_MASK);
+
+	return 0;
+}
+
+static int gpio_eos_s3_port_set_masked_raw(const struct device *dev,
+					   uint32_t mask,
+					   uint32_t value)
+{
+	ARG_UNUSED(dev);
+	uint32_t target_value;
+	uint32_t output_states = MISC_CTRL->IO_OUTPUT;
+
+	target_value = ((output_states & ~mask) | (value & mask));
+	MISC_CTRL->IO_OUTPUT = (target_value & GPIOS_MASK);
+
+	return 0;
+}
+
+static int gpio_eos_s3_port_set_bits_raw(const struct device *dev,
+					 uint32_t mask)
+{
+	ARG_UNUSED(dev);
+
+	MISC_CTRL->IO_OUTPUT |= (mask & GPIOS_MASK);
+
+	return 0;
+}
+
+static int gpio_eos_s3_port_clear_bits_raw(const struct device *dev,
+					   uint32_t mask)
+{
+	ARG_UNUSED(dev);
+
+	MISC_CTRL->IO_OUTPUT &= ~(mask & GPIOS_MASK);
+
+	return 0;
+}
+
+static int gpio_eos_s3_port_toggle_bits(const struct device *dev,
+					uint32_t mask)
+{
+	ARG_UNUSED(dev);
+	uint32_t target_value;
+	uint32_t output_states = MISC_CTRL->IO_OUTPUT;
+
+	target_value = output_states ^ mask;
+	MISC_CTRL->IO_OUTPUT = (target_value & GPIOS_MASK);
+
+	return 0;
+}
+
+static int gpio_eos_s3_manage_callback(const struct device *dev,
+				       struct gpio_callback *callback, bool set)
+{
+	struct gpio_eos_s3_data *data = dev->data;
+
+	return gpio_manage_callback(&data->callbacks, callback, set);
+}
+
+static int gpio_eos_s3_pin_interrupt_configure(const struct device *dev,
+					       gpio_pin_t gpio_num,
+					       enum gpio_int_mode mode,
+					       enum gpio_int_trig trig)
+{
+	struct gpio_eos_s3_data *data = dev->data;
+	const struct gpio_eos_s3_config *config = dev->config;
+	uint8_t is_secondary = (config->pin_secondary_config >> gpio_num) & 1;
+	GPIOCfgTypeDef gpio_cfg;
+	PadConfig pad_conf;
+
+	gpio_cfg.ucGpioNum = gpio_num;
+	pad_conf = pad_configs[(8 * is_secondary) + gpio_num];
+	gpio_cfg.xPadConf = &pad_conf;
+
+	if (mode == GPIO_INT_MODE_DISABLED) {
+		/* Reset IRQ configuration */
+		int irq_num = HAL_GPIO_IntrCfg(&gpio_cfg);
+
+		if (irq_num < 0) {
+			return -EINVAL;
+		}
+
+		data->gpio_irq_num[gpio_cfg.ucGpioNum] = BIT((uint8_t)irq_num);
+		/* Disable IRQ */
+		INTR_CTRL->GPIO_INTR_EN_M4 &= ~data->gpio_irq_num[gpio_num];
```

Comment:
You use `gpio_cfg.ucGpioNum` and `gpio_num` interchangeably. Let's maybe stick to `gpio_num`, the code will be easier to read.

Reply:
refactored, thanks